lib/abci/handlers/commitHandlerFactory.js

```diff
@@ -46,9 +50,12 @@ function commitHandlerFactory(
       await blockExecutionDBTransactions.commit();
 
       // Store ST fees from the block to distribution pool
-      chainInfo.setCreditsDistributionPool(blockExecutionContext.getAccumulativeFees());
+      creditsDistributionPool.setCreditsDistributionPool(
```

Comment:
We should use a transaction as well. You may find LevelDBTransaction in git history.

Reply:
Agreed to implement it in #429